src/processes/dryrun.js

```diff
@@ -0,0 +1,341 @@
+const ethers = require("ethers");
+const { Router } = require("sushi/router");
+const { visualizeRoute, getSpanException } = require("../utils");
+
+/**
+ * Specifies the reason that dryrun failed
+ */
+const DryrunHaltReason = {
+    NoOpportunity: 1,
+    NoWalletFund: 2,
+    NoRoute: 3,
+};
+
+/**
+ * Tries to find the maxInput for an arb tx by doing a binary search
+ */
+async function dryrun({
```

Comment:
does this replace something that already exists? i would expect to see some deletions that map to these additions

Reply:
this specific fn han no changes, but you can check the diff against src/processOrders.js dryrun fn

Reply:
also if I do the deletions, bunch of things will be broken, is that ok with you?

Reply:
also `dryrunWithRetries()` is a new fn which is refactor based on certain part of src/processOrders.js:processPair()